client/src/App.js

```diff
@@ -1,12 +1,45 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
 import GlobalStyle from "./GlobalStyle";
+import { LoadingScreen } from "./components/LoadingScreen";
+import { HomePage } from "./pages/HomePage";
+import { AddProjectPage } from "./pages/AddProjectPage";
+import { BrowsePage } from "./pages/BrowsePage";
+import { CategoryPage } from "./pages/CategoryPage";
+import { DetailsPage } from "./pages/DetailsPage";
+import { FavouritesPage } from "./pages/FavouritesPage";
 
 function App() {
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    setTimeout(() => setLoading(false), 4000);
+  }, []);
+
   return (
-    <>
+    <Router>
       <GlobalStyle />
-      <div className="App">Test deployment</div>
-    </>
+      <Switch>
+        <Route exact path="/">
+          {loading === false ? <HomePage /> : <LoadingScreen />}
```

Comment:
If loading is a bool you could try `{loading ? <LoadingScreen/> : <HomePage/>}`

Reply:
thanks for the tip!